### PR TITLE
feat: Swift strict concurrency

### PR DESCRIPTION
**Pull request by PhilippeWeidmann**

### CODE REVIEW

File: login-Example/login-Example/ViewController.swift [hunk 1]
```diff
@@ -117,53 +113,52 @@ class ViewController: UIViewController, InfomaniakLoginDelegate, DeleteAccountDe
...
+                            "UserId: \(token.userId)\nToken: \(token.accessToken)\nExpires in: \(token.expiresIn ?? -1)"
+                        self.testSwapRefreshToken(apiToken: token)
+                    } else if let error = error {
+                        title = "Login error"
+                        description = error.localizedDescription
                     }
+                    print("refreshTokenConvert \(title ?? "")\n\(description ?? "")")
+                }
+            case .failure(let failure):
+                DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
```
Comment (adrien-coye) [inline login-Example/login-Example/ViewController.swift:155]:
```suggestion
                Task {
            try? await Task.sleep(nanoseconds:1_000_000_000)
```

For the sake of using Swift's structured concurrency everywhere, we could do this instead.

**Review by adrien-coye — COMMENTED** [review]
I have two remarks, let me know what you think.

File: login-Example/login-Example/ViewController.swift [hunk 2]
```diff
@@ -57,26 +57,24 @@ class ViewController: UIViewController, InfomaniakLoginDelegate, DeleteAccountDe
...
-                                self.present(deleteAccountViewController, animated: true)
-                            }
+        loginService.asWebAuthenticationLoginFrom(anchor: ASPresentationAnchor(),
+                                                  useEphemeralSession: true,
+                                                  hideCreateAccountButton: true) { result in
+            switch result {
+            case .success((let code, let verifier)):
+                self.tokenService.getApiTokenUsing(code: code, codeVerifier: verifier) { token, _ in
+                    if let token = token {
+                        DispatchQueue.main.async {
```
[the remark this answers is not in the document]
Comment (PhilippeWeidmann) [reply login-Example/login-Example/ViewController.swift:67]: Good catch, didn't notice that we were using DispatchQueues in the sample project. Will fix.

File: login-Example/login-Example/ViewController.swift (hunk 1 above)
Comment (PhilippeWeidmann) [reply login-Example/login-Example/ViewController.swift]: I removed the delays. I don't know why they were there in the first place.

**Review by adrien-coye — APPROVED** [review]
Ty 🙌

**Pull request merged**